android/src/main/java/com/adyenreactnativesdk/component/AdyenCardComponent.kt

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) 2021 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ */
+package com.adyenreactnativesdk.component
+
+import com.facebook.react.bridge.ReactApplicationContext
+import com.facebook.react.bridge.ReactMethod
+import com.facebook.react.bridge.ReadableMap
+import com.adyenreactnativesdk.configuration.RootConfigurationParser
+import com.adyen.checkout.components.model.payments.Amount
+import com.adyenreactnativesdk.configuration.CardConfigurationParser
+import com.adyen.checkout.card.CardConfiguration
+import org.json.JSONException
+import android.util.Log
+import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.DialogFragment
+import com.adyen.checkout.card.CardComponent
+import com.adyen.checkout.card.CardView
+import com.adyen.checkout.components.model.payments.request.PaymentComponentData
+import com.adyen.checkout.components.ActionComponentData
+import com.adyen.checkout.components.model.payments.response.Action
+import com.adyen.checkout.core.api.Environment
+import com.adyenreactnativesdk.*
+import java.lang.Exception
+import java.util.*
+
+class AdyenCardComponent(context: ReactApplicationContext?) : BaseModule(context),
+    PaymentComponentListener, ActionHandlingInterface {
+    val TAG = "CardComponent"
+    val PAYMENT_METHOD_KEY = "scheme"
+    private var actionHandler: ActionHandler? = null
+    private var dialog: DialogFragment? = null
+
+    override fun getName(): String {
+        return "AdyenCardComponent"
+    }
+
+    @ReactMethod
+    fun open(paymentMethodsData: ReadableMap?, configuration: ReadableMap?) {
+        val paymentMethods = getPaymentMethodsApiResponse(paymentMethodsData)
+        val paymentMethod = getPaymentMethod(paymentMethods, PAYMENT_METHOD_KEY)
+
+        val config = RootConfigurationParser(configuration!!)
+        val environment: Environment
+        val clientKey: String
+        val shopperLocale: Locale
+        val amount: Amount?
+        try {
+            environment = config.environment
+            clientKey = config.clientKey
+            shopperLocale = config.locale
+            amount = config.amount
+        } catch (e: NoSuchFieldException) {
+            sendEvent(DID_FAILED, ReactNativeError.mapError(e))
+            return
+        }
+
+        val actionHandlerConfiguration = ActionHandlerConfiguration(shopperLocale, environment, clientKey)
+        actionHandler = ActionHandler(this, actionHandlerConfiguration)
+
+        val parser = CardConfigurationParser(configuration)
+        val componentConfiguration: CardConfiguration
+        val builder = CardConfiguration.Builder(shopperLocale, environment, clientKey)
+        componentConfiguration = parser.getConfiguration(builder)
+
+        val theActivity = appCompatActivity
+        val viewModel = ComponentViewModel(paymentMethod!!, shopperLocale, amount)
```

Comment:
Be careful with `!!`, it will throw an exception when the object is `null`. It is better to handle these cases safely

Reply:
Updated! Thanks :)